src/main/scala/cromwell/engine/backend/jes/JesBackend.scala

```diff
@@ -150,92 +150,98 @@ class JesBackend extends Backend with LazyLogging {
     )
   }
 
-  override def setupCallEnvironment(call: Call, workflowDescriptor: WorkflowDescriptor): JesTaskExecutionContext = {
-    val callGcsPath = s"${workflowDescriptor.callDir(call)}"
-    JesTaskExecutionContext(GoogleCloudStoragePath(callGcsPath), JesConnection)
+  override def bindCall(workflowDescriptor: WorkflowDescriptor,
+                        call: Call,
+                        locallyQualifiedInputs: CallInputs,
+                        abortRegistrationFunction: AbortFunctionRegistration): BackendCall = {
+    JesBackendCall(this, workflowDescriptor, call, locallyQualifiedInputs, abortRegistrationFunction)
   }
 
-  override def executeCommand(instantiatedCommandLine: String,
-                              workflowDescriptor: WorkflowDescriptor,
-                              call: Call,
-                              backendInputs: CallInputs,
-                              scopedLookupFunction: ScopedLookupFunction,
-                              abortFunctionRegistration: AbortFunctionRegistration): Try[Map[String, WdlValue]] = {
+  override def execute(backendCall: BackendCall): Try[CallOutputs] = backendCall match {
+    case bc: JesBackendCall => execute(bc)
```

Comment:
I did try what you suggested a while ago.  It got a lot more complicated than you might imagine.  I even got Miguel's eyes on it, we both realized it was a hard problem.  I tried a few things (covariant types, implicit class, type parameters on the execute method, type parameters on BackendType), and I couldn't get any of them to reasonably work.  I had to fall back to this when those attempts failed.  I can try one more time or perhaps this can be a further refactoring we can do later?